lib/active_job/queue_adapters/sneakers_adapter.rb

```diff
@@ -0,0 +1,39 @@
+module ActiveJob
+  module QueueAdapters
+    # Explicitly remove the implementation existing in older rails'.
+    remove_const(:SneakersAdapter) if defined?(:SneakersAdapter)
+
+    # = Sneakers adapter for Active Job
+    #
+    # To use Sneakers set the queue_adapter config to +:sneakers+.
+    #
+    #   Rails.application.config.active_job.queue_adapter = :sneakers
+    class SneakersAdapter < ::ActiveJob::QueueAdapters::AbstractAdapter
+      def initialize
+        @monitor = Monitor.new
+      end
+
+      def enqueue(job)
+        @monitor.synchronize do
+          JobWrapper.from_queue job.queue_name
+          JobWrapper.enqueue ActiveSupport::JSON.encode(job.serialize)
```

Comment:
Does this encoding preserves everything like key symbols in hash?

```
ActiveSupport::JSON.decode(ActiveSupport::JSON.encode({id: 1}))
=> {"id"=>1}
```

Is that acceptable? If I remember well, there was already some kind of serialization in active job maybe we can reuse in here.

Reply:
That is a good question. I basically just copy-over already working [sneakers/kicks adapter](https://github.com/rails/rails/blob/797790828bae932c41a353b30f799c179966840c/activejob/lib/active_job/queue_adapters/sneakers_adapter.rb) from rails thinking that it is already proven to work, didn't want to change anything.

Json decode is fedded by `json.serialize` that comes from [ActiveJob](https://github.com/rails/rails/blob/main/activejob/lib/active_job/core.rb#L107), so this line of code doesn't look wrong to me, but then again you know much more about kicks than me :)